scipy/signal/_arraytools.py

```diff
@@ -207,3 +207,37 @@ def const_ext(x, n, axis=-1):
                           right_ext),
                          axis=axis)
     return ext
+
+
+def zero_ext(x, n, axis=-1):
+    """
+    Zero padding at the boundaries of an array
+
+    Generate a new ndarray that is a zero padded extension of `x` along
+    an axis.
+
+    Parameters
+    ----------
+    x : ndarray
+        The array to be extended.
+    n : int
+        The number of elements by which to extend `x` at each end of the
+        axis.
+    axis : int, optional
+        The axis along which to extend `x`.  Default is -1.
+
+    Examples
+    --------
+    >>> from scipy.signal._arraytools import zero_ext
+    >>> a = np.array([[1, 2, 3, 4, 5], [0, 1, 4, 9, 16]])
+    >>> zero_ext(a, 2)
+    array([[ 0,  0,  1,  2,  3,  4,  5,  0,  0],
+           [ 0,  0,  0,  1,  4,  9, 16,  0,  0]])
+    """
+    if n < 1:
+        return x
+    zeros_shape = list(x.shape)
+    zeros_shape[axis] = n
+    zeros = np.zeros(zeros_shape, dtype=x.dtype)
+    ext = np.concatenate((zeros, x, zeros), axis=axis)
+    return ext
```

Comment:
Yeah, I wanted to create a function with the same syntax as `odd_ext` et al, so I could do the function dictionary thing in `_spectral_helper`.

I played around with reimplementing the `*_ext` functions as wrappers for `np.pad`, but in practice these simple implementations turn out to be a fair amount faster than the equivalent call to `np.pad` (probably since `pad` is set up to do arbitrary padding along multiple dimensions.)

Then again, I doubt that padding factors much into the overall computation time of `_spectral_helper`...